### PR TITLE
turn camel into snake #698

### PR DESCRIPTION
**Pull request by isaacJChen**

### CODE REVIEW

File: lib/core/versions/latest/DocumentComposer.ts [hunk 1]
```diff
@@ -124,35 +124,35 @@ export default class DocumentComposer {
    * Validates the schema of the given full document.
    * @throws SidetreeError if given document patch fails validation.
    */
-  public static validateDocument (document: any) {
+  private static validateDocument (document: any) {
```
Comment (isaacJChen) [inline lib/core/versions/latest/DocumentComposer.ts:127]: I turned this into private because I don't see it used anywhere lmk if this is not ok

Comment (thehenrytsai) [reply lib/core/versions/latest/DocumentComposer.ts:127]: Looks good here.

File: lib/core/versions/latest/DocumentComposer.ts [hunk 2]
```diff
@@ -124,32 +124,32 @@ export default class DocumentComposer {
...
 
-    const allowedProperties = new Set(['publicKeys', 'serviceEndpoints']);
+    const allowedProperties = new Set(['public_keys', 'service_endpoints']);
     for (let property in document) {
       if (!allowedProperties.has(property)) {
         throw new SidetreeError(ErrorCode.DocumentComposerUnknownPropertyInDocument, `Unexpected property ${property} in document.`);
       }
     }
 
     // Verify 'publicKeys' property if it exists.
```
Comment (thehenrytsai) [inline lib/core/versions/latest/DocumentComposer.ts:139]:
```suggestion
    // Verify 'public_keys' property if it exists.
```
or remove if you feel it's too trivial!

Same for all occurrences.

File: lib/core/versions/latest/DocumentComposer.ts [hunk 3]
```diff
@@ -124,32 +124,32 @@ export default class DocumentComposer {
...
+    if (document.hasOwnProperty('service_endpoints')) {
       // Verify each serviceEndpoint entry in serviceEndpoints.
-      DocumentComposer.validateServiceEndpoints(document.serviceEndpoints);
+      DocumentComposer.validateServiceEndpoints(document.service_endpoints);
     }
   }
 
   /**
-   * Validates the schema of the given update document patch.
+   * Parses and validates the schema of the given update document patch.
```
Comment (thehenrytsai) [inline lib/core/versions/latest/DocumentComposer.ts:152]: Nit^9000:

Technically just validate as nothing returned, probably lingered from commit.

File: tests/core/DocumentComposer.spec.ts [hunk 4]
```diff
@@ -233,19 +233,19 @@ describe('DocumentComposer', async () => {
       const patch = {
         extra: 'unknown value',
         action: 'add-service-endpoints',
-        serviceEndpoints: 'not an array'
+        service_endpoints: 'not an array'
       };
       const expectedError = new SidetreeError(ErrorCode.DocumentComposerPatchMissingOrUnknownProperty);
-      expect(() => { DocumentComposer['validateRemoveServiceEndpointsPatch'](patch); }).toThrow(expectedError);
+      expect(() => { DocumentComposer['validateAddServiceEndpointsPatch'](patch); }).toThrow(expectedError);
```
Comment (thehenrytsai) [inline tests/core/DocumentComposer.spec.ts:239]: Thanks!

**Pull request merged**